core/src/color.rs

```diff
@@ -88,6 +96,41 @@ impl Color {
         }
     }
 
+    /// Creates a [`Color`] from a hex string. Supported formats are #rrggbb, #rrggbbaa, #rgb,
+    /// #rgba. The “#” is optional. Both uppercase and lowercase are supported.
+    pub fn from_hex(s: &str) -> Result<Color, ColorError> {
+        let mut hex = s.replace('#', "").to_lowercase();
```

Comment:
Using `strip_prefix` is more safe choice, to not allow something like `###01#2##856`.

Also, the overall method seems overcomplicated, many unnecessary allocations. Just check length, then loop: take 1 or 2 characters (depending on the length check), use `from_str_radix()`

Reply:
You're totally right, thank you! Fixed.